perf/perf_24x7_counter.py

```diff
@@ -0,0 +1,201 @@
+        # Check if its enabled
+        # Refer https://bugzilla.linux.ibm.com/show_bug.cgi?id=139404#c21
+        result_perf = process.run("%s,domain = 2,core = 1/ sleep 1"
+                                  % perf_stat, ignore_status=True)
+        if "You may not have permission to collect stats" in\
+                result_perf.stderr:
+            raise self.error("Please enable lpar to allow\
+                collecting the 24x7 counters info")
+
+        # Testing feature: Display domain indices in sysfs
+        # Part of commit d34171e
+        #EVENT_SYSFS = "/sys/bus/event_source/devices/hv_24x7/events"
+        pattern = re.compile('1: Physical Chip\n2: Physical Core\n\
+            3: VCPU Home Core\n4: VCPU Home Chip\n\
+            5: VCPU Home Node\n6: VCPU Remote Node')
+        result = process.run('cat %s/interface/domains' % event_sysfs)
+        Result_search = pattern.search(result.stdout)
+        if Result_search:
+            self.log.info('Displayed domain indices in sysfs')
+        else:
+            nfail += 1
+            self.log.info('FAIL : Unable to display domain indices in sysfs')
+
+        # Testing feature: Eliminate domain suffix in event name
+        # Verify No __PHYS_CORE or VCPU* suffixes in event name
+        search_suffix = process.run('ls %s/events |grep -E\
+            __PHYS_CORE' % event_sysfs, ignore_status=True)
+        if search_suffix.stdout:
+            nfail += 1
+            self.log.info('FAIL : Found __PHYS_CORE  suffixes in event name')
+        else:
+            self.log.info('No __PHYS_CORE  suffixes in event name')
+
+        search_suffix = process.run('ls %s/events |grep\
+            -E VCPU' % event_sysfs, ignore_status=True)
+        if search_suffix.stdout:
+            nfail += 1
+            self.log.info('FAIL : Found VCPU  suffixes in event name')
+        else:
+            self.log.info('No VCPU  suffixes in event name')
+
+        # Using hv_24x7/HPM_0THRD_NON_IDLE_CCYC__PHYS_CORE should fail
+        result1 = process.run('%s__PHYS_CORE,core=1/ sleep\
+            1' % perf_stat, ignore_status=True)
+        if "Invalid event/parameter" not in result1.stdout:
+            nfail += 1
+            self.log.info('FAIL : perf unable to recognize hv_24x7/\
+                HPM_0THRD_NON_IDLE_CCYC__PHYS_CORE has invalid event')
+        else:
+            self.log.info('perf recognized Invalid event')
+
+        # Check if missing parameter is reported
+        result1 = process.run(
+            '%s/ sleep 1' % perf_stat, ignore_status=True)
+        if "invalid or unsupported event" not in result1.stderr or "Required\
+                parameter 'domain' not specified" not in result1.stdout:
+            nfail += 1
+            self.log.info('FAIL : Domain is not specified,\
+                perf unable to recognize it has invalid event')
+        else:
+            self.log.info('perf recognized unsupported event')
+
```

Comment:
can this 109 to 126 to also part of helper function ? and where it is applicable